app/celery/nightly_tasks.py

```diff
@@ -122,6 +123,24 @@ def delete_letter_notifications_older_than_retention():
         raise
 
 
+@notify_celery.task(name="clean-notifications-table-and-create-stats")
+@statsd(namespace="tasks")
+def clean_notifications_table_and_create_stats():
+    """
+    A group of functions to run sequentially as their order matters
+
+    We timeout any notifications that are still in 'sending'. When that is done we can then create
+    the rows in the notification_status table using the data from the notifications table.
+    Then we have finished interpreting the notifications table and can move notifications into the
+    notification_history table and delete the originals from `notifications`.
+    """
+    timeout_notifications()
+    create_nightly_notification_status()
```

Comment:
`create_nightly_notification_status` spawns a load of `create_nightly_notification_status_for_day` tasks. This new task won't control when those tasks run unless the entire thing is triggered after the delete tasks finish.

Those tasks all call `get_notification_table_to_use` to figure out if notifications have been deleted yet. That function has a flag for if the delete tasks have already run today so we can use that